entropylab/results_backend/sqlalchemy/db.py

```diff
@@ -1,416 +1,467 @@
-                    query.module,
-                    query.class_name,
-                    query.driver,
-                    query.version,
-                    query.driver_type,
-                    query.args,
-                    query.kwargs,
-                    cached_metadata["functions"],
-                    cached_metadata["parameters"],
-                    cached_metadata["undeclared_functions"],
-                )
-            else:
-                return None
-
-    def _get_driver_id(self, name) -> int:
-        with self._session_maker() as sess:
-            query = (
-                sess.query(Resources)
-                .filter(Resources.name == name)
-                .order_by(desc(Resources.update_time))
-                .first()
-            )
-            if query and not query.deleted:
-                return int(query.id)
-            else:
-                return -1
-
-    def set_locked(self, resource_name):
-        pass
-
-    def set_released(self, resource_name):
-        pass
-
-    def get_all_resources(self) -> Set[str]:
-        with self._session_maker() as sess:
-            query = (
-                sess.query(Resources)
-                .filter(Resources.deleted == False)  # noqa: E712
-                .group_by(Resources.name)
-                .all()
-            )
-            if query is not None:
-                return {item.name for item in query}
-            else:
-                return set()
+import logging
+from datetime import datetime
+from typing import List, TypeVar, Optional, ContextManager, Iterable, Union, Any
+from typing import Set
+
+import jsonpickle
+import pandas as pd
+from pandas import DataFrame
+from sqlalchemy import desc
+from sqlalchemy.exc import DBAPIError
+from sqlalchemy.orm import sessionmaker, Session
+from sqlalchemy.sql import Selectable
+from sqlalchemy.util.compat import contextmanager
+
+from config import settings
+from entropylab.api.data_reader import (
+    DataReader,
+    ExperimentRecord,
+    ResultRecord,
+    MetadataRecord,
+    DebugRecord,
+    PlotRecord,
+)
+from entropylab.api.data_writer import (
+    DataWriter,
+    ExperimentInitialData,
+    ExperimentEndData,
+    RawResultData,
+    Metadata,
+    Debug,
+    PlotSpec,
+    NodeData,
+)
+from entropylab.instruments.instrument_driver import Function, Parameter
+from entropylab.instruments.lab_topology import (
+    PersistentLabDB,
+    DriverType,
+    ResourceRecord,
+)
+from entropylab.results_backend.sqlalchemy.storage import HDF5Storage
+from entropylab.results_backend.sqlalchemy.db_initializer import _DbInitializer
+from entropylab.results_backend.sqlalchemy.lab_model import (
+    Resources,
+    ResourcesSnapshots,
+)
+from entropylab.results_backend.sqlalchemy.model import (
+    ExperimentTable,
+    PlotTable,
+    ResultTable,
+    DebugTable,
+    MetadataTable,
+    NodeTable,
+)
+
+T = TypeVar(
+    "T",
+)
+
+
+# noinspection PyBroadException
```

Comment:
Can we just narrow the exception type and not ignore lint errors?

Reply:
I've removed all #noinspection directives. Narrowed down broad except clauses to `RuntimeError`